File: src/test/smallTestProject/AlarmClock/Model/src/main/java/com/h2m/alarm/model/AlarmClock.java
package com.h2m.alarm.model;

import com.h2m.common.observer.Observable;

public final class AlarmClock extends Observable implements Runnable
{
    public final static String ALARM_EVENT = "alarm";

    static
    {
    }
    
    public AlarmClock()
    {
        super(ALARM_EVENT);
    }

    public void run()
    {
        for (int i = 0; i < 5; i++)
        {
            try
            {
                System.out.println("Tick");
                Thread.sleep(1000);
            }
            catch (InterruptedException e)
            {
                e.printStackTrace();
            }
        }

        notifyAboutEvent(ALARM_EVENT);
    }
}

File: src/test/NAlarmClock/Model/AlarmClock.cs
using System;
using System.Threading;

namespace Model
{
	public class AlarmClock : Foundation.Observable
	{
		public static String ALARM_EVENT = "alarm";

		public AlarmClock() : base(ALARM_EVENT)
		{
		}

		public void runIt() 
		{
			Thread newThread = new Thread(new ThreadStart(Run));
			newThread.Start(); 
		}

		public void Run()
		{
			for (int i = 0; i < 5; i++)
			{
				Console.Write("Tick");
				Thread.Sleep(1000);

			}

			notifyAboutEvent(ALARM_EVENT);
		}
	}
}

